Sources/LocheckLogic/ProblemReporter.swift

```diff
@@ -15,44 +15,94 @@ private struct StderrOutputStream: TextOutputStream {
     }
 }
 
+public enum Severity: String {
+    case warning
+    case error
+    case ignored
+}
+
+public protocol Problem {
+    var kindIdentifier: String { get }
+    var uniquifyingInformation: String { get }
+    var severity: Severity { get }
+    var message: String { get }
+}
+
+public protocol SummarizableProblem: Problem {
+    // Provide the string key, allowing errors to be grouped and displayed in a nice
+    // human-readable format
+    var key: String { get }
+}
+
 /**
  Collect problems found during a run, and log them to the console for Xcode integration if desired.
  */
 public class ProblemReporter {
-    public struct Problem: Error, Equatable {
-        public enum Severity: String {
-            case warning
-            case error
-        }
-
+    public struct LocalProblem: Error, Equatable {
         public let path: String
         public let lineNumber: Int
-        public let message: String
-        public let severity: Severity
+        public let problem: Problem
+
+        var messageForXcode: String {
+            "\(path):\(lineNumber): \(problem.severity.rawValue): \(problem.message) (\(problem.kindIdentifier))"
```

Comment:
Does Xcode show the error on line 1 if we print 0 for the line number here?

Reply:
No, but it opens the file. Xcode will not show per-line errors for plist files, apparently. That's one reason why I added the summary section, so it's easier for a human to browse the script output.